libs/core/src/main/java/org/elasticsearch/common/CharArrays.java

```diff
@@ -1,9 +1,22 @@
 /*
```

Comment:
im super weary of moving things that rest depends on to server. The hope is that we will depend even less on server, or not at all. Id rather see this moved into client if it *needs* to be, or if its just using one or two of those helpers, to just move those.

Reply:
Until we can remove the need for serialization, I think this should stay as is. Once we can split our internal serialization from the request/response objects then this becomes a non-issue. Currently we already depend on StreamInput and StreamOutput, which pulls in this dependency and lucene too.